File: numeric/euclidgcd/euclidgcd.erl
-module(euclidgcd).
-export([main/0, euclidgcd/2, print_all/0]).

euclidgcd(M, 0) -> M;
euclidgcd(M, N) -> euclidgcd(N, M rem N).

print_all() ->
    receive
        {M, N} -> io:format("~p ~p~ngcd: ~p~n", [M, N, euclidgcd(M, N)])
    end.

main() ->
    P = spawn(euclidgcd, print_all, []),
    Args = init:get_plain_arguments(),
    case Args of
        [Arg1Str, Arg2Str] ->
            P ! {list_to_integer(Arg1Str), list_to_integer(Arg2Str)};
        _Other ->
            P ! {15, 10}
    end,
    ok.
    

File: numeric/euclidgcd/euclidgcd.java
package numeric.euclidgcd;

public class euclidgcd
{    
    public static int perform(int m, int n)
    {
        int r = 0;
        while (n != 0) 
        {
            r = m % n;
            m = n;
            n = r;
        }
        return m;
    }

    public static void main(String[] args)
    {
        int m = 15;
        int n = 10;

        if (args.length >= 2)
        {
            m = Integer.parseInt(args[0]);
            n = Integer.parseInt(args[1]);
        }

        System.out.format("%d %d\ngcd: %d\n", m, n, perform(m, n));
    }
}
